cmake/Platform.cmake

```diff
@@ -22,4 +24,15 @@ if (MSVC)
     # Suppress warnings: "Prefer enum class over enum" (Enum.3)
 
     add_compile_options(/wd4267 /wd26812)
+
+    # warning level 3 and all warnings as errors
+    add_compile_options(/W3 /WX)
+
+    # enable parallel builds
+    add_compile_options(/MP)
```

Comment:
This I am not in favor of because it makes assumptions about the cmake generator being used.  For parallel builds, depending on the generator, you might use syntax such as:

```bash
cmake --build . --parallel 8
cmake --build . -- -j8
cmake --build . -- /m
cmake --build . -- /mp
etc.
```

It's generator-specific how to ask for a parallel build.

Reply:
Just add `-DCMAKE_CXX_FLAGS="/MP /W3 /Wx"` to your build.  There's no reason to push that out to everyone else who may or may want to use these flags.   I think keeping thing minimal is better.